tokio/src/task/scope/intrusive_double_linked_list.rs

```diff
@@ -0,0 +1,628 @@
+//! An intrusive double linked list of data
```

Comment:
Kind of amusing to note that if we merge this PR as-is, tokio would now contain at least 4 separate linked-list implementations. :)

Reply:
Am I correct that this linked list is inherently not thread safe? This should probably be stated in this comment...

Reply:
Re 4 list implementations: Sounds indeed not ideal :-)
I confess I didn't perform an in-depth review on what is available up to now, since the main focus was to get `scope` working. This one here was mostly copy/paste to enable the necessary synchronization primitives.

Re thread-safety: It indeed is not. I thought not being `Sync` is indicator enough :-)